src/rascal/math/bessel.hh

```diff
@@ -222,19 +282,29 @@ namespace rascal {
        *         Note that a reference is returned to avoid unnecessary
        *         copies.
        */
-      auto get_values() { return ref(bessel_values); }
+      auto get_values() {
+        if (this->compute_gradients) {
+          // when the gradient are computed bessel_values has one additional
+          // column that should not be returned
```

Comment:
Could the extra column be stored separately?

Reply:
it could but the implementation of gradient_recursion, upward_recursion and downward_recursion would not be as straight forward.

Reply:
ok, fair enough.